process/block/baseProcess.go

```diff
@@ -976,6 +949,29 @@ func (bp *baseProcessor) removeHeadersBehindNonceFromPools(
 	}
 }
 
+func (bp *baseProcessor) removeBlockBodyOfHeader(headerHandler data.HeaderHandler) error {
+	if bp.onRequestBlockBodyOfHeader == nil {
```

Comment:
this verification is not needed.

Reply:
instead of function you may use the interface.

Reply:
done